tests/integ/modin/io/test_to_csv.py

```diff
@@ -0,0 +1,178 @@
+        path_expected: Path to file with expected output.
+        is_compressed: True if outfile files are gzip compressed.
+    """
+
+    def read_file_content(path: str) -> str:
+        if is_compressed:
+            with gzip.open(path) as f:
+                return f.read().decode()
+        else:
+            with open(path) as f:
+                return f.read()
+
+    assert_equal(read_file_content(path_actual), read_file_content(path_expected))
+
+
+def get_filepaths(kwargs: Any, test_name: str) -> Tuple[str, str]:
+    ext = kwargs.get("ext", "csv")
+    native_path = os.path.join(TEMP_DIR_NAME, f"native_{test_name}.{ext}")
+    snow_path = os.path.join(TEMP_DIR_NAME, f"snow_{test_name}.{ext}")
+    # Remove files if exits.
+    if os.path.exists(native_path):
+        os.remove(native_path)
+    if os.path.exists(snow_path):
+        os.remove(snow_path)
+    return native_path, snow_path
+
+
+@pytest.mark.parametrize("kwargs, is_compressed", SERIES_TEST_CASES)
+@sql_count_checker(query_count=1)
+def test_to_csv_series_local(kwargs, is_compressed):
+    native_series = native_pd.Series(["one", None, "two"], name="A")
+    native_path, snow_path = get_filepaths(kwargs, "series_local")
+
+    kwargs = kwargs.copy()
+    kwargs.pop("ext", None)
+    # Write csv with native pandas.
+    native_series.to_csv(native_path, **kwargs)
+    # Write csv with snowpark pandas.
+    pd.Series(native_series).to_csv(snow_path, **kwargs)
+
+    assert_file_equal(snow_path, native_path, is_compressed)
+
+
+@pytest.mark.parametrize("kwargs, is_compressed", DATAFRAME_TEST_CASES)
+@sql_count_checker(query_count=1)
+def test_to_csv_dataframe_local(kwargs, is_compressed):
+    native_df = native_pd.DataFrame({"A": ["one", "two", None], "B": [1, 2, 3]})
+    native_path, snow_path = get_filepaths(kwargs, "dataframe_local")
+
+    kwargs = kwargs.copy()
+    kwargs.pop("ext", None)
+    # Write csv with native pandas.
+    native_df.to_csv(native_path, **kwargs)
+    # Write csv with snowpark pandas.
+    pd.DataFrame(native_df).to_csv(snow_path, **kwargs)
+
+    assert_file_equal(snow_path, native_path, is_compressed)
+
+
+@pytest.mark.parametrize("kwargs, is_compressed", SERIES_TEST_CASES)
+@sql_count_checker(query_count=2)
+def test_to_csv_series_stage(sf_stage, session, kwargs, is_compressed):
+    native_series = native_pd.Series(["one", None, "two"], name="A")
+    # None index name is not supported when writing to snowflake stage.
+    native_series.index.name = "X"
+    native_path, snow_path = get_filepaths(kwargs, "series_stage")
+
+    kwargs = kwargs.copy()
+    kwargs.pop("ext", None)
+    # Write csv with native pandas.
+    native_series.to_csv(native_path, **kwargs)
+    # Write csv to snowflake stage.
+
+    stage_location = f"@{sf_stage}/{os.path.basename(snow_path)}"
+    if kwargs.get("index", True) is False:
+        # Writing single column with null values has different behavior in snowpark
```

Comment:
why not match pandas behavior?

Reply:
Maybe the pandas behavior is a bug. I filed https://github.com/pandas-dev/pandas/issues/59116

Reply:
Yeah, this behavior seems very off to me. Thanks for filing the bug.

Reply:
Could you add a link to the pandas bug here? I'm still not sure whether it's a bug, but we can revisit the behavior later.

Reply:
Added link to bug.